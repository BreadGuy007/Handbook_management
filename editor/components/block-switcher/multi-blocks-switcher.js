/**
 * External dependencies
 */
import { connect } from 'react-redux';

/**
 * Internal dependencies
 */
import './style.scss';
import BlockSwitcher from './';
import { getMultiSelectedBlockUids } from '../../store/selectors';

export function MultiBlocksSwitcher( { isMultiBlockSelection, selectedBlockUids } ) {
	if ( ! isMultiBlockSelection ) {
		return null;
	}
	return (
		<BlockSwitcher key="switcher" uids={ selectedBlockUids } />
	);
}

export const mapStateToProps = ( state ) => {
	const selectedBlockUids = getMultiSelectedBlockUids( state );
	return {
		isMultiBlockSelection: selectedBlockUids.length > 1,
		selectedBlockUids,
	};
};

export default connect(
	mapStateToProps
)( MultiBlocksSwitcher );
