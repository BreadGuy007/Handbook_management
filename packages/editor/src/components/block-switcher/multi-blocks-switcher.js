/**
 * WordPress dependencies
 */
import { withSelect } from '@wordpress/data';

/**
 * Internal dependencies
 */
import BlockSwitcher from './';

export function MultiBlocksSwitcher( { isMultiBlockSelection, selectedBlockClientIds } ) {
	if ( ! isMultiBlockSelection ) {
		return null;
	}
	return (
		<BlockSwitcher key="switcher" clientIds={ selectedBlockClientIds } />
	);
}

export default withSelect(
	( select ) => {
		const selectedBlockClientIds = select( 'core/editor' ).getMultiSelectedBlockClientIds();
		return {
			isMultiBlockSelection: selectedBlockClientIds.length > 1,
			selectedBlockClientIds,
		};
	}
)( MultiBlocksSwitcher );
