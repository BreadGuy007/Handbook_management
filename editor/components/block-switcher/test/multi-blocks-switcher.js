/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import { MultiBlocksSwitcher } from '../multi-blocks-switcher';

describe( 'MultiBlocksSwitcher', () => {
	test( 'should return null when the selection is not a multi block selection.', () => {
		const isMultiBlockSelection = false;
		const selectedBlockClientIds = [
			'clientid',
		];
		const wrapper = shallow(
			<MultiBlocksSwitcher
				isMultiBlockSelection={ isMultiBlockSelection }
				selectedBlockClientIds={ selectedBlockClientIds }
			/>
		);

		expect( wrapper.html() ).toBeNull();
	} );

	test( 'should return a BlockSwitcher element matching the snapshot.', () => {
		const isMultiBlockSelection = true;
		const selectedBlockClientIds = [
			'clientid-1',
			'clientid-2',
		];
		const wrapper = shallow(
			<MultiBlocksSwitcher
				isMultiBlockSelection={ isMultiBlockSelection }
				selectedBlockClientIds={ selectedBlockClientIds }
			/>
		);

		expect( wrapper ).toMatchSnapshot();
	} );
} );
