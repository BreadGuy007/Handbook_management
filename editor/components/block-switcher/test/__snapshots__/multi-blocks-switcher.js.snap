// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiBlocksSwitcher should return a BlockSwitcher element matching the snapshot. 1`] = `
<WrappedComponent
  uids={
    Array [
      "an-uid",
      "another-uid",
    ]
  }
/>
`;
